File: Expressions/Functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, grocieries, transportation, savings, expenses, spend;

float  input(char type[], float var){
    printf("Monthly %s:\n" , type);
    scanf( "%.2f" , &var);
    return var;
}


void percent(char type[], int amount){
    int per = amount/income *100;

    printf("Your %s is %d%% of your income. \n" , type, per);
}


int main(void){
printf("This is a budget calculator. \n How much do you make a month? \n");
income = input("income", income);
rent = input("rent", rent);
utilities = input("utilities", utilities);
grocieries = input("grocieries", grocieries);
transportation = input("transportation", transportation);
savings = income * .2;
income = input("income", income);
expenses = rent + utilities +grocieries +transportation;
spend = income - expenses - savings;
printf("You make $%.2f\n" , income);
printf("Your expenses are $%.2f\n" , expenses);
printf("Your savings are $%.2f\n" , savings);
printf("Your spending money is $%.2f\n", spend);

percent("rent", rent); 
percent("utilities", utilities); 
percent("grocieries", grocieries); 
percent("transportaion", transportation); 
percent("expenses", expenses); 
percent("savings", savings); 
percent("spending", spend); 
return 0;
}

File: Expressions/Functions/update_budget.py



def get_financials(type):

    money = float(input(f"What is your {type}? \n"))

    return money


def cal_per(type, amount,income):
    percent = amount/income *100

    return f"Your {type} is {percent} "

income = get_financials("income")
rent = get_financials('rent') 
utilities = get_financials("utilities")
grocieries = get_financials("grocieries")
transportation = get_financials("transportation")
savings = get_financials("savings")
expenses = get_financials("expenses")


print(cal_per("rent %", rent, income))
print(cal_per("utilities %", utilities, income))
print(cal_per("grocieries %", grocieries, income))
print(cal_per("transoportation %",transportation,income))
print(cal_per("savings %", savings, income))
print(cal_per("rent %", rent, income))

